include/gamepad/joystick_transformation.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include <concepts>
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace gamepad {
+
+/**
+ * @brief An abstract class for joystick transformations. 
+ * 
+ * A transformation takes a coordinate representing the value of the joystick, and returns a transformed coordinate 
+ * value
+ * 
+ */
+class AbstractTransformation {
+    public:
+    /**
+     * @brief Get the transformed coordinate given the original.
+     * 
+     * @param original The original value of the joystick
+     * @return std::pair<float, float> The transformed value
+     */
```

Comment:
this should be aligned with the function. Usually clang-format catches this

```suggestion
      /**
       * @brief Get the transformed coordinate given the original.
       * 
       * @param original The original value of the joystick
       * @return std::pair<float, float> The transformed value
       */
```